lock.go

```diff
@@ -114,29 +135,34 @@ func (l *Locker) create() (bool, error) {
 			return true, nil
 		}
 
-		if time.Now().Add(l.opts.WaitRetry).After(stop) {
-			break
+		if attempts--; attempts <= 0 {
+			return false, nil
 		}
 
-		if l.opts.RetriesCount > 0 && retries <= 0 {
-			break
+		if retryDelay == nil {
+			retryDelay = time.NewTimer(l.opts.RetryDelay)
+			defer retryDelay.Stop()
+		} else {
+			retryDelay.Reset(l.opts.RetryDelay)
 		}
 
-		retries--
-		time.Sleep(l.opts.WaitRetry)
+		select {
+		case <-ctx.Done():
```

Comment:
it always will wait and only then check `ctx.Done()`

Reply:
sorry, what?

Reply:
look.. if you want to break with context, nothing happens. The whole cycle is performed in any case at least once (in the worst case - will be granted the lock and returned it) and only then code is run `select {}`

Reply:
```go
select {
case <-ctx.Done():
  return false, ctx.Err()
case <-retryDelay.C:
}
```

in this case, if we doesn't receive anything here (`case <-ctx.Done()`) we run wait (`case <-retryDelay.C`) and go try again

Reply:
https://play.golang.org/p/qUIlgmWhNd

Reply:
and see this - https://play.golang.org/p/RkqQXbM50l (ticker + context)

Output
```
tick 0
Done
```

Reply:
@im-kulikov that's by design. the context is just there to allow us to terminate the sleep. Assume you set a `RetryCount: 10, RetryDelay: time.Second`. in the worst case it will try 11x with 10s delay in between. Assuming you want to `Close()` whatever your lock is part of, you can just terminate the root context which will cause the last `LockWithContext(root)` earlier than 10s. 

It's also in implicit replacement for `WaitTimeout` for those people who have very specific Lock deadline requirements.

Please let me know if there's anything else, will merge otherwise.

Reply:
if I don't convince you on this point, then only https://github.com/bsm/redis-lock/pull/7#discussion_r150554507

Reply:
sorry, that's really an absolutely OK thing to do. will merge